Win GDI+

OgreTransporter opened the pull request:
I have created a GDI+ map renderer (https://docs.microsoft.com/en-us/windows/win32/gdiplus/-gdiplus-gdi-start) with a demo.

This is useful for low-level WinAPI programming and the use of Windows frameworks like MFC.

Framstag commented:
Great news :-)

The is currently a linker on msys...

You should also add it to: http://libosmscout.sourceforge.net/documentation/directory/

please tell me, if I should merge...(you can use the new draft function of github for this, too).

OgreTransporter commented:
I'm finished so far. With the documentation I also adapted the website.

Karry commented on CMakeLists.txt:
```diff
@@ -276,6 +278,15 @@ if(OSMSCOUT_BUILD_MAP_DIRECTX)
 endif()
+if(OSMSCOUT_BUILD_MAP_GDI)
+  if(NOT ${OSMSCOUT_BUILD_MAP})
+    message(SEND_ERROR "The main map drawing interface is required for DirectX map drawing backend")
```
`DirectX` in the message is not correct I guess

Karry commented on libosmscout-map-gdi/src/osmscout/MapPainterGDI.cpp:
```diff
@@ -0,0 +1,764 @@
+
+		void Release()
+		{
+			for (std::map<Gdiplus::ARGB, Gdiplus::SolidBrush*>::iterator iter = m_SolidBrushes.begin(); iter != m_SolidBrushes.end(); iter++) delete iter->second;
+			for (std::map<PENDEF, Gdiplus::Pen*>::iterator iter = m_Pens.begin(); iter != m_Pens.end(); iter++) delete iter->second;
+			for (std::map<FONTDEF, Gdiplus::Font*>::iterator iter = m_Fonts.begin(); iter != m_Fonts.end(); iter++) delete iter->second;
+			for (std::map<size_t, Gdiplus::Image*>::iterator iter = m_Images.begin(); iter != m_Images.end(); iter++) delete iter->second;
```
it would be nice to use for-range loops this old style is verbose

Karry reviewed (commented):
Thank you for that massive work! I just add few suggestions. Also, it would be great to add it that new backend to table at http://libosmscout.sourceforge.net/features/rendering/

But as you see, there is problem with rendering this page :-( I will look at it...

OgreTransporter commented:
@Karry: Oops! Copy and paste error in CMake ;-) I have also adjusted the loops. I also added the GDI renderer to the website. All changes are in the new PR #954.
